enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -2438,6 +2438,26 @@ Function *CreatePrimalAndGradient(
 
   gutils->eraseFictiousPHIs();
 
+  if (topLevel) {
+    IRBuilder<> entryBuilder(gutils->inversionAllocs,
+                             gutils->inversionAllocs->begin());
+    bool seenshared = false;
+    for (auto &g : gutils->newFunc->getParent()->globals()) {
+      if (hasMetadata(&g, "enzyme_internalshadowglobal")) {
+        entryBuilder.CreateStore(Constant::getNullValue(g.getValueType()), &g);
+        if (g.getType()->getAddressSpace() == 3) {
```

Comment:
```suggestion
        if (g.getType()->getAddressSpace() == llvm::ADDRESS_SPACE_SHARED) {
```

Reply:
Apparently that's inside of the LLVM lib and thus may not be installed, so we can't use :'( 

```
wmoses@beast:~/git/Enzyme/enzyme/build10 (cudasync) $ find ../../ -iname NVPTXBaseInfo.h
../../flang/llvm/lib/Target/NVPTX/MCTargetDesc/NVPTXBaseInfo.h
../../llvm/lib/Target/NVPTX/MCTargetDesc/NVPTXBaseInfo.h
```

Reply:
sadness